### PR TITLE
Solution to week 3's mini challenges

### PR DESCRIPTION
**Pull request by Danotsonof**
@bondz, Kindly approve my PR.

### CODE REVIEW

File: src/binary-reversal/index.js [hunk 1]
```diff
@@ -3,6 +3,11 @@
  *
  *  * @param {string} value
  */
-function binaryReversal(value) {}
+function binaryReversal(value) {
+	let paddedbinary = parseInt(value).toString(2).padStart(8,0)
+	let reversedBinary = paddedbinary.split('').reverse().join('')
+	let decimal = parseInt(reversedBinary,2)
```
Comment (bondz) [inline src/binary-reversal/index.js:9]: Nice. These can be `const` though.

File: src/list-sorting/index.js [hunk 2]
```diff
@@ -1,3 +1,14 @@
-function listSorting(needle, haystack) {}
+function listSorting(needle, haystack) {
+	if (Array.isArray(haystack[0])) {
+		for (let i = haystack.length-1; i >= 0; i--) {
```
Comment (bondz) [inline src/list-sorting/index.js:3]: Doesn't always have to be `i`, can be `row`

File: src/binary-reversal/index.js (hunk 1 above)
Comment (Danotsonof) [reply src/binary-reversal/index.js:9]: true, thanks.

File: src/list-sorting/index.js (hunk 2 above)
Comment (Danotsonof) [reply src/list-sorting/index.js:3]: 💯  adding that to my memory bank. Thanks

**Pull request merged**